src/Orchard.Environment.Shell/ShellStateUpdater.cs

```diff
@@ -52,7 +53,11 @@ public class ShellStateUpdater : IShellStateUpdater
             var shellState = await _stateManager.GetShellStateAsync();
 
             // start with description of all declared features in order - order preserved with all merging
-            var orderedFeatureDescriptors = _extensionManager.GetExtensions().Features;
+            var allUnorderedFeatures = _extensionManager.GetExtensions().Features.ToArray();
```

Comment:
Or another idea, just have an `_extensionManager.GetEnabledFeaturesAsync()` method that returns all enabled features for the current tenant, already ordered, and it can even cache it. No need to expose LoadFeatureAsync, or to have to deal with OrderByDependenciesAndPriorities (which should be private).